src/mixing_fab_tests.cc

```diff
@@ -0,0 +1,386 @@
+  TEST_F(MixingFabTest, InitialState) {
+    // Test things about the initial state of the facility here
+    std::vector<std::string> in_com_ = { "in_c1", "in_c2", "in_c3" };
+    std::vector<double> in_frac_ = { 0.80, 0.15, 0.05 };
+    std::vector<double> in_cap_ = { 30, 20, 10 };
+    
+    SetStream_comds(in_com_);
+    SetStream_ratio(in_frac_);
+    SetStream_capacity(in_cap_);
+    
+    SetOutStream_comds("out_com");
+    SetOutStream_capacity(50);
+    
+    SetThroughput(1e200);
+    
+    
+    std::vector<std::string> strm_comds_ = GetStream_comds();
+    std::vector<double> strm_ratio_ = GetStream_ratio();
+    std::vector<double> strm_cap_   = GetStream_capacity();
+    
+    for (int i = 0; i < in_com.size(); i++) {
+      EXPECT_EQ(in_com[i],  strm_comds_[i]);
+      EXPECT_EQ(in_frac[i], strm_ratio_[i]);
+      EXPECT_EQ(in_cap[i],  strm_cap_[i]);
+    }
+    
+    EXPECT_EQ(out_com, GetOutStream_comds());
+    EXPECT_EQ(out_cap, GetOutStream_capacity());
+    
+    EXPECT_EQ(throughput, GetThroughput());
+      }
+
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+  TEST_F(MixingFabTest, StreamRatio) {
+    // Test things about the mixing ratio of the facility here
+    
+    std::vector<std::string> in_com_ = { "in_c1", "in_c2", "in_c3" };
+    std::vector<double> in_cap_ = { 30, 20, 10 };
+    
+    SetStream_comds(in_com_);
+    SetStream_capacity(in_cap_);
+    
+    SetOutStream_comds("out_com");
+    SetOutStream_capacity(50);
+    
+    SetThroughput(1e200);
+    
+    mf_facility_->EnterNotify();
+    double ext_val = 1/3.;
+    std::vector<double> strm_ratio_ = GetStream_ratio();
+    for (int i = 0; i < in_com.size(); i++) {
+      EXPECT_DOUBLE_EQ(ext_val, strm_ratio_[i]);
+    }
+    
+    
+    
+    in_com_ = { "in_c1", "in_c2", "in_c3" };
+    std::vector<double> in_frac_ = { 2, 1, 5 };
+    in_cap_ = { 30, 20, 10 };
+    
+    SetStream_comds(in_com_);
+    SetStream_ratio(in_frac_);
+    SetStream_capacity(in_cap_);
+    
+    SetOutStream_comds("out_com");
+    SetOutStream_capacity(50);
+    
+    SetThroughput(1e200);
+    
+    mf_facility_->EnterNotify();
+    
+    strm_ratio_ = GetStream_ratio();
+    double sum = 0;
+    for (int i = 0; i < in_com.size(); i++) {
+      sum += strm_ratio_[i];
+    }
+    
+    EXPECT_DOUBLE_EQ(sum, 1) << "Normalisation of the mising ratio went wrong!";
+    
+    in_com_ = { "in_c1", "in_c2", "in_c3" };
+    in_frac_ = { 0.1, 0.2, 0.5 };
+    in_cap_ = { 30, 20, 10 };
+    
+    SetStream_comds(in_com_);
+    SetStream_ratio(in_frac_);
+    SetStream_capacity(in_cap_);
+    
+    SetOutStream_comds("out_com");
+    SetOutStream_capacity(50);
+    
+    SetThroughput(1e200);
+    
+    mf_facility_->EnterNotify();
+    
+    strm_ratio_ = GetStream_ratio();
+    sum = 0;
+    for (int i = 0; i < in_com.size(); i++) {
+      sum += strm_ratio_[i];
+    }
+    
+    EXPECT_DOUBLE_EQ(sum, 1) << "Normalisation of the mising ratio went wrong!";
```

Comment:
This can be inside the same test as the above normalization check, with the same improved message - but some doc comments mentioning the difference between the two checks would be nice.